### ISSUE TITLE
Artist search list not showing full list of users in gallery / Searching Feature?

### ISSUE THREAD

**Issue by RandomFluffyUser** [SpottedSqueak/FA-Gallery-Downloader#17]
Hi, I am up to date and this has been working well, and im not sure if this is intentional or a mishap on my part but is it normal when opening the gallery to not be able to see all users from a-z? I recently downloaded my favorites and a ton of artists popped up but when you look in the autofill for sorting by artist for me It only goes to "P" as shown in the picture
![Screenshot 2024-08-09 193707](https://github.com/user-attachments/assets/833fd263-0674-4e5b-bb66-ac3b1f2f5eb6)
And to set it straight I can still search for users beyond the list, for example lets say I wanted to see the pieces made by user "Xaphelm" that I favorited, I can still search for and see the pieces that i favorited, I just do not know I favorited some of their work and saved it because the autocomplete stops after a while. Unless this is a technical issue its not too much of a problem for me currently.

As well as something else I wanted to add, I am not sure if its possibble but perhaps a better way to distinguish sorting? Unless it requires to redownload all my galleries I was curious if instead of the search bar searching the description/title/tags all at once, perhaps a way to distinguish them? I do hope im not asking too much, just wanted to throw this idea out there.

**Comment by SpottedSqueak** [comment]
Curious, what OS is this? I'm afraid that dropdown list is limited by the OS's interpretation of the component, so there's not much I can do there. I could make it an actual select box, but you wouldn't be able to search with a select box, they'd have to be different controls.

Also, you'll have to clarify that second point. What are you asking for? An option to limit search to specific attributes (title, tags, etc.) only?

**Comment by RandomFluffyUser** [comment]
I am on windows 11! Like I said its not too bothersome, it just intrigued me since after downloading my favorites so many artists popped up that it kind of messed with the full list. I was thinking differentiating between artists from favorites and main uploads but I presume that would be quite difficult so there's no concern about that for me for that.

And as in for my second point I was unsure if it was possible but I saw in the gallery whenever you search it appears to search for everything, whether it be in the description, the title, or the tags. So I was thinking perhaps a slightly more refined option to let you search instead for words in the description, tags, and title separately? Again not sure if it is possible but it would be greatly appreciated.

**Comment by SpottedSqueak** [comment]
Haha, yeah, there's a limit to how many lines can be shown in that box, unfortunately. I could probably add in a separate select box that contains *all* of those artists too. Just that the more controls I add there, the more complex the database queries get.

As for the proposed feature...I might be able to do something like keyword queries? Kinda like how they do it on FA and other places... "@ tags [tags listed here]", things like that. That gets complex too, depending! I'll noodle on this a bit more.

**Comment by RandomFluffyUser** [comment]
Yeah I get that for the first option, it was more of a little QOL thing but I don't really _need it,_ other than that I will stay updated with this and can't wait to see what you can do for a possible keyword system!
